server/mediaserver.go

```diff
@@ -393,20 +387,61 @@ func (s *LivepeerServer) registerConnection(rtmpStrm stream.RTMPVideoStream) (*r
 		profile:     &vProfile,
 		params:      params,
 		sessManager: NewSessionManager(s.LivepeerNode, params, playlist),
+		lastUsed:    time.Now(),
 	}
+
 	s.connectionLock.Lock()
 	s.rtmpConnections[mid] = cxn
 	s.lastManifestID = mid
 	s.lastHLSStreamID = hlsStrmID
 	sessionsNumber := len(s.rtmpConnections)
 	s.connectionLock.Unlock()
+
+	ticker := getConnectionTicker()
+	go func(s *LivepeerServer, mid core.ManifestID) {
+		for _ = range ticker.C {
+			lastUsed := s.rtmpConnections[mid].lastUsed
+			now := time.Now()
+			diff := now.Sub(lastUsed)
+			if diff.Hours() >= 1 || diff.Minutes() >= 1 {
+				err := removeRTMPStream(s, mid)
+				if err != nil {
+					continue
```

Comment:
Let's just stop the ticker here regardless of whether an error occurs.

Otherwise, the the ticker loops indefinitely if an error occurs. Since we're not sure such errors are actually recoverable, this is tantamount to a memory leak.

This would currently happen with the RTMP ingest workflow:

* RTMP comes in, cxn registers, ticker is kicked off
* RTMP disconnects, cxn is cleaned up
* Ticker ticks, finds no RTMP connection, returns error
* Ticker ticks, finds no RTMP connection, returns error
* ...

(As noted earlier, we probably don't need to actually start the cleanup goroutine for RTMP ingest. Doing so could be useful one day to resume a HLS session after a RTMP disconnect, but that's outside the scope of this PR.)

Reply:
Thanks for the detailed explanation! Is this what you're suggesting? If `removeRTMPStream` errors and cxn isn't cleaned up, wouldn't that be a memory leak if ticker is stopped and cleanup isn't retried?

```golang
if time.Since(lastUsed) > connectionRefreshInterval {
     _ = removeRTMPStream(s, mid)
    ticker.Stop()
    return
}
```

Reply:
>  If removeRTMPStream errors and cxn isn't cleaned up, wouldn't that be a memory leak if ticker is stopped and cleanup isn't retried?

That would be true if the error was somehow recoverable by the next tick. Don't think we have any cases like that.

The code sample looks good. Could probably omit the `_ =` from `removeRTMPStream`.